### PR TITLE
Add Values.podLabels and Values.podAnnotations

### PR DESCRIPTION
**Pull request by vostres**
Add Values.podLabels and Values.podAnnotations to:

producer-app

producer-app-clean-up-job

streams-app-clean-up-job

### CODE REVIEW

File: charts/producer-app-cleanup-job/templates/job.yaml [hunk 1]
```diff
@@ -15,19 +15,19 @@ spec:
   ttlSecondsAfterFinished: 30
   template:
     metadata:
-    {{- if or .Values.awsRole .Values.annotations }}
```
Comment (philipp94831) [inline charts/producer-app-cleanup-job/templates/job.yaml:18]: annotations are not used anymore. Either we reuse them from the values or we add them to the job metadata (see above, labels are there). Any opinion @torbsto ?

File: charts/streams-app-cleanup-job/templates/job.yaml [hunk 2]
```diff
@@ -15,19 +15,19 @@ spec:
   ttlSecondsAfterFinished: 30
   template:
     metadata:
-    {{- if or .Values.awsRole .Values.annotations }}
```
Comment (philipp94831) [inline charts/streams-app-cleanup-job/templates/job.yaml:18]: same comment as above

File: charts/producer-app/templates/pod.yaml [hunk 3]
```diff
@@ -1,19 +1,19 @@
 {{- define "producer-app.podTemplate" -}}
 {{- $root := . -}}
 metadata:
-{{- if or .Values.awsRole .Values.annotations }}
```
Comment (philipp94831) [inline charts/producer-app/templates/pod.yaml:4]: same comment as above

**Review by philipp94831 — CHANGES_REQUESTED** [review]
Please update READMEs

File: charts/producer-app-cleanup-job/templates/job.yaml (hunk 1 above)
Comment (philipp94831) [reply charts/producer-app-cleanup-job/templates/job.yaml:18]: I think we should just add them for consistency

File: charts/producer-app/templates/pod.yaml (hunk 3 above)
Comment (vostres) [reply charts/producer-app/templates/pod.yaml:4]: > same comment as above

producer-app already has Values.annotations in deployment.yaml. 
but there are only Values.labels in job.yaml. Should I add it there?

Comment (philipp94831) [reply charts/producer-app/templates/pod.yaml:4]: yes please

File: charts/streams-app/templates/deployment.yaml [hunk 4]
```diff
@@ -48,17 +48,17 @@ spec:
         prometheus.io/scrape: "true"
         prometheus.io/port: {{ .Values.prometheus.jmx.port | quote }}
       {{- end }}
-      {{- range $key, $value := .Values.annotations }}
+      {{- range $key, $value := .Values.podAnnotations }}
         {{ $key | quote }}: {{ $value | quote }}
       {{- end }}
-      {{- if and .Values.autoscaling.consumerGroup (not .Values.annotations.consumerGroup) }}
+      {{- if and .Values.autoscaling.consumerGroup (not .Values.podAnnotations.consumerGroup) }}
```
Comment (philipp94831) [inline charts/streams-app/templates/deployment.yaml:54]: We don't actually need the consumer group annotation on the pod, only on the deployment, so you can remove it here

**Comment by philipp94831** [comment]
Can you please also update rclone-copy chart?

File: charts/streams-app/templates/deployment.yaml [hunk 5]
```diff
@@ -48,7 +48,7 @@ spec:
         prometheus.io/scrape: "true"
         prometheus.io/port: {{ .Values.prometheus.jmx.port | quote }}
       {{- end }}
-      {{- range $key, $value := .Values.annotations }}
+      {{- range $key, $value := .Values.podAnnotations }}
         {{ $key | quote }}: {{ $value | quote }}
       {{- end }}
       {{- if and .Values.autoscaling.consumerGroup (not .Values.annotations.consumerGroup) }}
```
Comment (philipp94831) [inline charts/streams-app/templates/deployment.yaml:54]: I meant removing this whole 3 line if statement :)

File: charts/streams-app/templates/deployment.yaml [hunk 6]
```diff
@@ -39,7 +39,7 @@ spec:
       release: {{ .Release.Name }}
   template:
     metadata:
-    {{- if or (or (or .Values.awsRole .Values.prometheus.jmx.enabled) .Values.autoscaling.consumerGroup) .Values.annotations }}
+    {{- if or (or (or .Values.awsRole .Values.prometheus.jmx.enabled) .Values.autoscaling.consumerGroup) .Values.podAnnotations }}
```
Comment (philipp94831) [inline charts/streams-app/templates/deployment.yaml:42]: remove the consumerGroup condition along with the next comment

**Review by philipp94831 — CHANGES_REQUESTED** [review]

File: charts/producer-app-cleanup-job/templates/job.yaml [hunk 7]
```diff
@@ -4,6 +4,12 @@ apiVersion: batch/v1
 kind: Job
 metadata:
   name: {{ template "producer-app-cleanup-job.fullname" . }}
+{{ - if .Values.annotations }}
```
Comment (philipp94831) [inline charts/producer-app-cleanup-job/templates/job.yaml:7]:
```suggestion
{{- if .Values.annotations }}
```

does that cause the issues?

File: charts/streams-app/templates/deployment.yaml (hunk 6 above)
Comment (vostres) [reply charts/streams-app/templates/deployment.yaml]: > remove the consumerGroup condition along with the next comment

I changed it, but here I am not 100% sure, if I got the condition right

Comment (philipp94831) [reply charts/streams-app/templates/deployment.yaml]: Looks good! (and much much better with the single or)

**Review by philipp94831 — CHANGES_REQUESTED** [review]
Please merge main

**Pull request merged**